pages/k8s/benchmarking.md

```diff
@@ -0,0 +1,169 @@
+---
+wrapper_template: "kubernetes/docs/base_docs.html"
+markdown_includes:
+  nav: "kubernetes/docs/shared/_side-navigation.md"
+context:
+  title: "Benchmarking"
+  description: How to benchmark a Charmed Kubernetes cluster
+keywords: benchmarking
+tags: [operating]
+sidebar: k8smain-sidebar
+permalink: benchmarking.html
+layout: [base, ubuntu-com]
+toc: False
+---
+
+The **Center for Internet Security (CIS)** maintains a
+[Kubernetes benchmark][cis-benchmark] that is helpful to ensure clusters are
+deployed in accordance with security best practices. **Charmed Kubernetes**
+includes support for the [kube-bench][] utility, which reports how well a
+cluster complies with the benchmark.
+
+## Run kube-bench
+
+The `kubernetes-master`, `kubernetes-worker`, and `etcd` charms used by
+**Charmed Kubernetes** include a `cis-benchmark` action that will install,
+configure, and run the benchmark on respective components. Run this action
+on the units you wish to test with the following:
+
+```bash
+juju run-action --wait etcd/0 cis-benchmark
+```
+
+By default, the action will display a summary of any issues found as well as
+the command that was executed on the unit. A `report` command is included
+to facilitate transferring the full benchmark report to a local machine for
+analysis.
+
+```yaml
+results:
+  cmd: /home/ubuntu/kube-bench/kube-bench -D /home/ubuntu/kube-bench/cfg --version
+    1.13-snap-etcd --noremediations --noresults master
+  report: juju scp etcd/0:/home/ubuntu/kube-bench-results/results-text-49681_7h .
+  summary: |
+    == Summary ==
+    7 checks PASS
+    0 checks FAIL
+    0 checks WARN
+    4 checks INFO
+status: completed
+```
+
+## Configure kube-bench
+
+The following parameters can be adjusted to change the default action behavior.
+See the descriptions in the [actions.yaml][layer-cis-benchmark-config] for
```

Comment:
@evilnick I took a shortcut here and just linked to the source of parameter options.  This keeps us from having to keep multiple places in sync, but it means the user has to go away to see all the available options.  Do you think it would be better to duplicate those options here for completeness in this doc?

Reply:
This will certainly do for now, but I think long term we will want to at least add some examples of values we:
 * have high confidence will always be there
 * are common things users will want to do